sdk/tests/core/test_az_hfsm_pipeline.c

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <azure/core/internal/az_hfsm_pipeline_internal.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/az_result.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+static az_hfsm_pipeline az_hfsm_pipeline_test;
+
+static az_hfsm_policy P01; // Outbound
+static az_hfsm_policy P02;
+static az_hfsm_policy P03; // Inbound
+
+typedef enum
+{
+  P_OUTBOUND_0 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 5),
+  P_INBOUND_0 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 6),
+  S_INBOUND_0 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 7),
+  S_INBOUND_1 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 8),
+  S_INBOUND_2 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 9),
+  S_INBOUND_3 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 10),
+  S_OUTBOUND_0 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 11),
+  S_OUTBOUND_1 = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 12)
+} test_az_hfsm_pipeline_event_type;
+
+static az_hfsm_event poutbound0_evt = { P_OUTBOUND_0, NULL };
+static az_hfsm_event pinbound0_evt = { P_INBOUND_0, NULL };
+static az_hfsm_event sinbound0_evt = { S_INBOUND_0, NULL };
+static az_hfsm_event sinbound1_evt = { S_INBOUND_1, NULL };
+static az_hfsm_event sinbound2_evt = { S_INBOUND_2, NULL };
+static az_hfsm_event sinbound3_evt = { S_INBOUND_3, NULL };
+static az_hfsm_event soutbound0_evt = { S_OUTBOUND_0, NULL };
+static az_hfsm_event soutbound1_evt = { S_OUTBOUND_1, NULL };
+
+static int refp01root = 0;
+static int refp02root = 0;
+static int refp03root = 0;
+static int timeout0 = 0;
+static int timeout1 = 0;
+#ifdef TRANSPORT_MQTT_SYNC
+static int loopout0 = 0;
+static int loopin0 = 0;
+#endif
+static int poutbound0 = 0;
+static int pinbound0 = 0;
+static int sinbound0 = 0;
+static int sinbound1 = 0;
+static int sinbound2 = 0;
+static int sinbound3 = 0;
+static int soutbound0 = 0;
+static int soutbound1 = 0;
+
+static az_result P01Root(az_hfsm* me, az_hfsm_event event)
```

Comment:
nit here and elsewhere. the style of the embedded sdk is lower case underscore.
```suggestion
static az_result p_01_root(az_hfsm* me, az_hfsm_event event)
```